modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/consensus/ValidationErrorStorage.scala

```diff
@@ -0,0 +1,50 @@
+package io.constellationnetwork.node.shared.infrastructure.consensus
+
+import java.time.Instant
+
+import cats.data.NonEmptyList
+import cats.effect.kernel.{Async, Clock, Ref}
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import eu.timepit.refined.auto.autoUnwrap
+import eu.timepit.refined.types.all.PosInt
+
+trait ValidationErrorStorage[F[_], Artifact, ValidationError] {
+  def addValidationError(artifact: Artifact, errors: ValidationError): F[Unit]
+  def addValidationErrors(errors: NonEmptyList[(Artifact, ValidationError)]): F[Unit]
+  def allValidationErrors: F[Vector[ValidationErrorStorageEntry[Artifact, ValidationError]]]
+}
+
+case class ValidationErrorStorageEntry[Artifact, ValidationError](artifact: Artifact, validationError: ValidationError, timestamp: Instant)
+
+object ValidationErrorStorage {
+
+  def make[F[_]: Async, Artifact, ValidationError](
+    maxSize: PosInt
+  ): F[ValidationErrorStorage[F, Artifact, ValidationError]] =
+    for {
+      validationErrorsR <- Ref[F].of(Vector.empty[ValidationErrorStorageEntry[Artifact, ValidationError]])
+    } yield
+      new ValidationErrorStorage[F, Artifact, ValidationError] {
+        override def addValidationError(artifact: Artifact, error: ValidationError): F[Unit] =
+          for {
+            timestamp <- Clock[F].realTimeInstant
```

Comment:
do you ever match errors for equality? how the timestamp is gonna by used? those timestamps are local timestamps and won't match any other timestamp in the network